lib/ansible/cli/galaxy.py

```diff
@@ -1451,13 +1460,18 @@ def execute_list_collection(self, artifacts_manager=None):
                     display.vvv("No collections found at {0}".format(collection_path))
                     continue
 
-                # Display header
-                fqcn_width, version_width = _get_collection_widths(collections)
-                _display_header(collection_path, 'Collection', 'Version', fqcn_width, version_width)
+                if output_format is not None:
```

Comment:
I think that the first branch of `if collection_name` should also emit JSON. Also, with Jordan's suggestion to use the `choices` kwarg in the CLI declaration, it'd be something like `output_format == 'json'`.

Reply:
Not sure I'm following. We loop over `collections_search_paths`. We need to include all of these in the json that's emitted, so we can only do that once we have the entirety of the data that needs to be printed.

Reply:
Just think about `ansible-galaxy collection list ns.name -o yaml`. Currently, your logic skips this case and falls back to human-readable format.